### ISSUE TITLE
Optimizing loading series to different yAxis's?

### ISSUE THREAD

**Issue by Silveress** [whawker/react-jsx-highcharts#146]
Hi there,

Loving the package and its doing almost all that I want.
However I am running into a performance issue that I cannot seem to overcome.

The goal of what I am doijg is to have as many items on the page as possible.
Each item has 12 series attached to it spread across 4 yAxis.

First yAxis loads up near instantly, with the following 3 far far slower.

I have tried adding the yaxis as props and the series also as props to see if that would help, however even doing it that way kept giving me Highcharts Error 18, axis not found.

Example in action: https://gw2.silveress.ie/history?dates=2012-08-26,2018-08-05&series=b,bvol,bval,bvalp&ids=19721,19697,43773,19699,19703,19698,19702,19700,19701
Source code: https://gitlab.com/Silveress_Gw2/Stats_Frontend/blob/5bf031f81101daa94da57704900c7c906b5a363c/src/history.js#L812

What advice can you give as regards the series and the axis?
With thanks.

**Comment by whawker** [comment]
Hi @Silveress looks like there is a high degree of latency whilst awaiting the data from the server - in excess of 5 seconds on the example you linked (see below)

<img width="1314" alt="screen shot 2018-08-07 at 21 17 34" src="https://user-images.githubusercontent.com/2003804/43800227-b23c8098-9a87-11e8-9201-5322feae4c7a.png">

I also think the functions you're using to format the data before passing to the chart could do with optimisation, there are lots of nested for loops, and merging of data. I would suggest using [array methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) instead.

Finally I think your `onSetExtremes` on the `XAxis` in causing some sort of infinite loop, looks to me as if Highcharts Boost is repeatedly loading image blobs, which suggests to me something is causing the chart to constantly re-render.

Here is an example chart, it is loading the same amount of data points, across 4 axes and 12 series, and it seems to load without any issues.

With mock 5 second API latency: [Demo](https://base-chart-example-mncbjt.stackblitz.io/), [Code](https://stackblitz.com/edit/base-chart-example-mncbjt?file=MyChart.js)
With no mock API latency: [Demo](https://base-chart-example-vnyejs.stackblitz.io/), [Code](https://stackblitz.com/edit/base-chart-example-vnyejs?file=MyChart.js)

**Comment by Silveress** [comment]
First of all thank you for the time.

I played around a lot with it last night and although yer examples didn't solve it for me they did point me in the right direction, namely that you only loaded 12 series.
In the example above I was loading 9 ID's  for a total of 112 series (4+(12*n)), turns out highcharts dosent like anything over 50 series or so.
The solution I went with was to only load the subset of series requested to teh hart.

As regards the latency, the api it is calling is my own one that I host on a  2.5 eur VPS, not overly powerful but gets the job done.

Again thank you for the help.


TLDR for anyone else: Highcharts can deal with a heck of a lot of data in a series but it dosent like 100+ series.
